testcafe/tests/pages/photos-model.js

```diff
@@ -61,20 +70,32 @@ export default class Page {
 
   async waitForLoading() {
     await t.expect(this.loading.exists).notOk('Page still loading')
+    await isExistingAndVisibile(this.contentWrapper, 'Content Wrapper')
   }
 
   //@param {string} when : text for console.log
+  //photoCount still failed sometimes, so let's try twice
   async getPhotosCount(when) {
-    let allPhotosCount
+    await checkAllImagesExists()
+    let allPhotosCount = await this.tryGetPhotosCount(when)
+    //try a second time to defined allPhotosCount is 1st try failed
+    if (typeof allPhotosCount === 'undefined') {
+      allPhotosCount = await this.tryGetPhotosCount(when)
```

Comment:
Can you explain why and when we can have `undefined` first?

Reply:
Changes : `getPhotosCount` now only works when there are images expected on page. Otherwise allphotCOunt is set to '0' manually, so we can use `t.expect` while counting